WalletWasabi/Blockchain/TransactionBroadcasting/TransactionBroadcaster.cs

```diff
@@ -93,6 +98,74 @@ public async Task<BroadcastingResult> BroadcastAsync(SmartTransaction tx, Cancel
 	}
 }
 
+public class ExternalTransactionBroadcaster : IBroadcaster
+{
+	public static readonly ImmutableArray<ExternalBroadcasterInfo> Providers =
+	[
+		new("BlockstreamInfo", ("https://blockstream.info", "http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion"), "/api/tx"),
+		new("MempoolSpace", ("https://mempool.space", "http://mempoolhqx4isw62xs7abwphsq7ldayuidyx2v2oethdhhj6mlo2r6ad.onion"), "/api/tx")
+	];
+
+	public static readonly ImmutableArray<ExternalBroadcasterInfo> TestNet4Providers =
+	[
+		new("MempoolSpace", ("https://mempool.space/testnet4", "http://mempoolhqx4isw62xs7abwphsq7ldayuidyx2v2oethdhhj6mlo2r6ad.onion/testnet4"), "/api/tx")
+	];
+
+	public ExternalTransactionBroadcaster(string providerName, Network network, IHttpClientFactory httpClientFactory)
+	{
+		if (network == Network.Main)
+		{
+			Broadcaster = Providers.FirstOrDefault(x => x.Name.Equals(providerName, StringComparison.InvariantCultureIgnoreCase)) ?? throw new NotSupportedException($"Transaction broadcaster '{providerName}' is not supported");
+		}
+		else
+		{
+			// TODO: Rework Full Node integration for Regtest, find more Testnet4 provider
+			Broadcaster = TestNet4Providers.First();
+		}
+
+		HttpClientFactory = httpClientFactory;
+		_userAgentGetter = UserAgent.GenerateUserAgentPicker(false);
+	}
+
+	public IHttpClientFactory HttpClientFactory { get; }
+
+	private UserAgentPicker _userAgentGetter;
+
+	private ExternalBroadcasterInfo Broadcaster { get; init; }
+
+	public async Task<BroadcastingResult> BroadcastAsync(SmartTransaction tx, CancellationToken cancellationToken)
+	{
+		Logger.LogInfo($"Trying to broadcast transaction via '{Broadcaster.Name}' API. TxID: {tx.GetHash()}.");
+		try
+		{
+			Uri requestUri = HttpClientFactory is OnionHttpClientFactory ? new Uri(Broadcaster.ApiDomain.Onion) : new Uri(Broadcaster.ApiDomain.ClearNet);
+
+			using var httpClient = HttpClientFactory.CreateClient($"{Broadcaster.Name}-{tx.GetHash()}");
+			httpClient.DefaultRequestHeaders.TryAddWithoutValidation("User-Agent", _userAgentGetter());
+			using var content = new StringContent($"{tx.Transaction.ToHex()}", Encoding.UTF8, "application/json");
+
+			using var response = await httpClient.PostAsync($"{requestUri}{Broadcaster.ApiEndpoint}", content, cancellationToken).ConfigureAwait(false);
+			response.EnsureSuccessStatusCode($"Error broadcasting tx {tx.GetHash()} to {Broadcaster.Name}");
```

Comment:
maybe could add a _Transaction is successfully broadcast {txID} by {Broadcast.Name} API_ log?

Reply:
Yes, good idea. Added in https://github.com/WalletWasabi/WalletWasabi/pull/13743/commits/43d0649913523dc8e0d73bcca3ec1b5417b64396

Reply:
imo _broadcast by 'BlockstreamInfo' API_ would be sufficient, instead of current:
```
INFO	TransactionBroadcaster.ReportSuccessfulBroadcast (337)	Transaction is successfully broadcast txID by BroadcastByExternalParty { ExternalApiName = BlockstreamInfo }.
```

Reply:
Yep, I messed up. Obviously your suggestion is more than sufficient.